src/main/java/org/fcrepo/auth/webac/FedoraWebACUserSecurityContext.java

```diff
@@ -0,0 +1,118 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.auth.webac;
+
+import java.security.Principal;
+import java.util.Arrays;
+
+import org.fcrepo.auth.common.FedoraAuthorizationDelegate;
+import org.fcrepo.auth.common.FedoraUserSecurityContext;
+
+import org.modeshape.jcr.ExecutionContext;
+import org.modeshape.jcr.value.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The security context for Fedora WebAC servlet users. These users are not
+ * necessarily authenticated by the container, i.e. users may include the
+ * general public. This security context delegates all access decisions to the
+ * configured authorization delegate.
+ *
+ * @author mohideen
+ */
+public class FedoraWebACUserSecurityContext extends FedoraUserSecurityContext {
+
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(FedoraWebACUserSecurityContext.class);
+
+    /**
+     * Constructs a new security context.
+     *
+     * @param userPrincipal the user principal associated with this security
+     *        context
+     * @param fad the authorization delegate
+     */
+    protected FedoraWebACUserSecurityContext(final Principal userPrincipal,
+                                             final FedoraAuthorizationDelegate fad) {
+        super(userPrincipal, fad);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @see org.modeshape.jcr.security.SecurityContext#hasRole(String)
+     */
+    @Override
+    public final boolean hasRole(final String roleName) {
+        // Under this custom PEP regime, all users have modeshape read and write
+        // roles.
+        if ("http://www.w3.org/ns/auth/acl#Read".equals(roleName)) {
+            return true;
+        } else if ("http://www.w3.org/ns/auth/acl#Write".equals(roleName)) {
+            return true;
+        } else if ("http://www.w3.org/ns/auth/acl#Append".equals(roleName)) {
+            return true;
+        } else if ("http://www.w3.org/ns/auth/acl#Control".equals(roleName)) {
+            return true;
+        }
+        return false;
+    }
+
+
+    /**
+     * Get the user principal associated with this context.
+     *
+     * @return the user principal associated with this security context
+     */
+    public Principal getEffectiveUserPrincipal() {
+        if (isLoggedIn() && this.userPrincipal != null) {
+            return this.userPrincipal;
+        }
+        return fad.getEveryonePrincipal();
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see
+     * org.modeshape.jcr.security.AdvancedAuthorizationProvider#hasPermission
+     * (org.modeshape.jcr.security.AdvancedAuthorizationProvider.Context,
+     * org.modeshape.jcr.value.Path, java.lang.String[])
+     */
+    @Override
+    public boolean hasPermission(final Context context, final Path absPath,
```

Comment:
Is this implementation of `hasPermission()` any different than:
https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-auth-common/src/main/java/org/fcrepo/auth/common/FedoraUserSecurityContext.java#L134

If not, please omit it from here.